tinyagentos/routes/agent_auth_requests.py

```
@@ -324,20 +340,62 @@ async def _do_approve(request: Request, request_id: str, body: ApproveBody, user
+        # them to 'active' so they are NOT in the bus inactive/revocation feed and
+        # @taOSmd's identity-AND-grant gate accepts them.
+        await registry.set_status(canonical_id, "active", actor=user.user_id)
+    except IntegrityError:
+        # A concurrent approve already took this active handle (the partial
+        # unique index fired). Roll back the failed write and remove the
+        # half-registered pending row so we never leave an active-without-handle
+        # agent or a stale pending row. Return the same friendly 409.
+        try:
+            await registry.rollback()
+        except Exception:  # noqa: BLE001 - never mask the 409 below
+            pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)